src/Reflection/Adapter/ReflectionClass.php

```diff
@@ -68,6 +70,22 @@ public function __toString()
         return $this->betterReflectionClass->__toString();
     }
 
+    /**
+     * @param string $name
+     *
+     * @return mixed
+     *
+     * @phpcsSuppress SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingNativeTypeHint
+     */
+    public function __get($name)
+    {
+        if ($name === 'name') {
+            return $this->betterReflectionClass->getName();
+        }
+
+        return $this->{$name};
```

Comment:
I think we need to prevent this - makes properties public-ish by default

Reply:
I haven't realized this. As there are no other public properties we probably don't need to care about that? But I'm not sure how to trigger the same error PHP itself does when stumbling upon an undefined property...

Reply:
`$this->betterReflectionClass`

Reply:
Just updated it.